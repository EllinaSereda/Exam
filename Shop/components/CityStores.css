.CityStores{
    float:right;
    width:80%;
   
}
.CityStores:after{
    clear:both;
    display: block;
    content:"";
}
.CityStores label {
    margin:10px 0;
    display:block;
    font-size:18px;
    
}
.CityStores .left{
    float:left;
}
@media screen and (orientation:portrait) and  (max-width:736pt) {
    .CityStores{
        margin-top:50px;
        width:100%;
        
    }
    .CityStores label {
        font-size:30px;
    }
  }