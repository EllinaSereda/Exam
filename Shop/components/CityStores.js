import React from 'react';
import PropTypes from 'prop-types';
import StoreInfo from './StoreInfo';
import './CityStores.css'
class CityStores extends React.PureComponent {

  static propTypes = {
    info:PropTypes.shape(
        {
          code:PropTypes.number,
          name:PropTypes.String,
          shops:PropTypes.arrayOf(
            PropTypes.shape({
                adress:PropTypes.String,
                code:PropTypes.number,
                tel:PropTypes.array,
                time:PropTypes.array,
                img:PropTypes.String,})
              
          ),
        } 
      )
  }

  state={
    checked:100, //выбранный адрес
  }

  show=(EO)=>{
    this.setState({checked:EO.target.value})
  }



  render() {
    console.log('CityStores did Render');
    let code=this.props.info.shops.map(v=>{ //массив с адресаними магазинов
    return <div key={v.code}><input  onChange={this.show} checked={v.code==this.state.checked?true:false} type="radio"  hidden value={v.code} id={v.code} name='filial'/>
    <label htmlFor={v.code}>{v.adress}</label></div> 
    });


    return <div className="CityStores"> 
     <div class="left">{code}</div>
     <StoreInfo info={this.props.info.shops.filter(v=>v.code==this.state.checked)[0]}/> 
    </div>  //Блок с инофрмацией о магазине: фото, режим работы

  }

}

export default CityStores;
