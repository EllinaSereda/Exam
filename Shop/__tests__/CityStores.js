"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import CityStores from '../components/CityStores';

test('работа CityStores', () => {

  // создаём тестовую версию компонента
  const component = renderer.create(
   <CityStores info={{code:0,name:'a',shops:[{adress:'адрес1',code:100,tel:['1','2'],time:['m','t'],img:['#1','#2']},
   {adress:'адрес2',code:102,tel:['1','2'],time:['m','t'],img:['#1','#2']}]}}/>
  );


  // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot();


});
