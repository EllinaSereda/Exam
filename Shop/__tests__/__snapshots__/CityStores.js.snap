// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`работа CityStores 1`] = `
<div
  className="CityStores"
>
  <div
    class="left"
  >
    <div>
      <input
        checked={true}
        hidden={true}
        id={100}
        name="filial"
        onChange={[Function]}
        type="radio"
        value={100}
      />
      <label
        htmlFor={100}
      >
        адрес1
      </label>
    </div>
    <div>
      <input
        checked={false}
        hidden={true}
        id={102}
        name="filial"
        onChange={[Function]}
        type="radio"
        value={102}
      />
      <label
        htmlFor={102}
      >
        адрес2
      </label>
    </div>
  </div>
  <div
    className="StoreInfo"
  >
    <img
      src={
        Array [
          "#1",
          "#2",
        ]
      }
    />
    <div>
      <span>
        Режим работы:
      </span>
       
      <p>
        m
      </p>
      <p>
        t
      </p>
    </div>
    <div>
      <span>
        Телефон:
      </span>
       
      <p>
        1
      </p>
      <p>
        2
      </p>
    </div>
  </div>
</div>
`;
